lib/puppet/provider/s3_bucket/v2.rb

```diff
@@ -39,4 +53,13 @@ def destroy
     s3_client.delete_bucket({bucket: name})
   end
 
+  def policy=(value)
+    Puppet.info('Replacing bucket policy')
+    Puppet.info(value.class)
```

Comment:
These lines should probably be combined into one (or perhaps remove the second?) and be `debug` not `info`